packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -3812,9 +3817,11 @@ namespace Ifpack2 {
       // wrap the workspace with 3d view
       vector_type_3d_view pmv(work.data(), num_blockrows, blocksize, num_vectors);
       const auto XX = X.getLocalViewDevice(Tpetra::Access::ReadOnly);
-      const auto YY = Y.getLocalViewDevice(Tpetra::Access::ReadWrite);
-      const auto ZZ = Z.getLocalViewDevice(Tpetra::Access::ReadWrite);
-      if (is_y_zero) Kokkos::deep_copy(YY, zero);
+      if (is_y_zero)
+      {
+        const auto YY = Y.getLocalViewDevice(Tpetra::Access::OverwriteAll);
+        Kokkos::deep_copy(YY, zero);
+      }
```

Comment:
I recommend to leave YY and ZZ as they are. I guessed that you changed this because the new tpetra code triggers host/device alive error. The following sequence of the code should run entirely on device. If there is something that prevents it, it means that we do unnecessary host view copy. It will cause a performance problem. We probably need to figure out where a local host view becomes necessary. We can talk offline about this.